libraries/AP_IBus_Telem/AP_IBus_Telem.cpp

```diff
@@ -0,0 +1,193 @@
+                    struct protocol_command_discover_response_t {
+                        uint8_t command_length;
+                        uint8_t address;
+                        uint16_t checksum;
+                    } packet {
+                        PROTOCOL_FOUR_LENGTH,
+                        (uint8_t)(PROTOCOL_COMMAND_DISCOVER | adr)
+                    };
+                    populate_checksum((uint8_t*)&packet, sizeof(packet));
+                    _port->write((uint8_t*)&packet, sizeof(packet));
+                    break;
+                }
+                case PROTOCOL_COMMAND_TYPE: { // 0x90, send sensor type
+                    // echo sensortype command: 0x06 0x91 0x00 0x02 0x66 0xFF
+                    struct protocol_command_type_response_t {
+                        uint8_t command_length;
+                        uint8_t address;
+                        uint8_t type;
+                        uint8_t length;
+                        uint16_t checksum;
+                    } packet {
+                        PROTOCOL_SIX_LENGTH,
+                        (uint8_t)(PROTOCOL_COMMAND_TYPE | adr),
+                        s->sensor_type,
+                        s->sensor_length
+                    };
+                    populate_checksum((uint8_t*)&packet, sizeof(packet));
+                    _port->write((uint8_t*)&packet, sizeof(packet));
+                    break;
+                }
+                case PROTOCOL_COMMAND_VALUE: { // 0xA0, send sensor data
+                    struct protocol_command_2byte_value_response_t {
+                        uint8_t command_length;
+                        uint8_t address;
+                        uint16_t value;
+                        uint16_t checksum;
+                    } packet {
+                        PROTOCOL_SIX_LENGTH,
+                        (uint8_t)(PROTOCOL_COMMAND_VALUE | adr),
+                        (uint16_t)s->sensor_value
+                    };
+                    populate_checksum((uint8_t*)&packet, sizeof(packet));
+                    _port->write((uint8_t*)&packet, sizeof(packet));
+                    break;
+                }
+                default:
+                    adr = 0; // unknown command, prevent sending chksum
+                    break;
+                }
+            }
+            break;
+        case DISCARD:
+            break;
+        default:
+            break;
+        }
+    }
+}
+
+void AP_IBus_Telem::setSensorMeasurements()
+{
+    //no need to do things if IBusTelemetry is not activated
+    if (_initialized) {
+        //sensors can easily be added if needed
+        uint16_t voltage = (uint16_t)(AP::battery().voltage() * 100);
+        _sensors[0].sensor_value =  voltage; //IBUS_MEAS_TYPE_EXTV
+        _sensors[1].sensor_value = AP::arming().is_armed(); //IBUS_MEAS_TYPE_ARMED
+        _sensors[2].sensor_value = (uint8_t)AP::vehicle()->get_mode(); //IBUS_MEAS_TYPE_FLIGHT_MODE
```

Comment:
I'm not going to ask you to change this, but I have a feeling that when we take this to a devcall that people will ask you to NOT update these values continuously, rather just grab them as needed.

After all - the requests don't come in *that* often, and it will save us both flash and RAM to not store them.